w3d_translator/w3d_translator.py

```diff
@@ -1,10 +1,154 @@
-from unity import UNITY_VERSION, UNITY_PATH
+import typer
+import glob
+import os
 
-# Opening Message
-def greeting():
-    print("W3D TRANSLATOR")
-    print(f"Unity Version:\t {UNITY_VERSION}")
-    print(f'Please ensure Unity is installed at {UNITY_PATH}\n')  
+from unity import (
+    UNITY_VERSION,
+    UNITY_PATH,
+    # create_project,
+    # copy_files,
+    # add_empty_scene,
+)
+from validate import validate_project, validate_in_multiple, validate_out
+from errors import ValidationError, TranslationError, UnityError
+from xml_to_unity import xml_to_unity
 
-if __name__ == '__main__':
-    greeting()
\ No newline at end of file
+# TODO: Configure typer character length (100)
+# TODO: Configure color/style for typer.echo
+# TODO: Prompt user for confirmation if --force is used (?)
+
+# TEMPORARY OVERRIDES (for development)
+# Skip validate_out
+# Skip create_project and copy_files (xml_to_unity.py)
+# Only translate run.xml (xml_to_unity.py)
+
+""" TODO NEXT
+Beginning translating Story (separate PRs)
+"""
+
+is_error = False
+
+
+# Opening message
+def greeting(in_dir: str, out_dir: str):
+    typer.echo("W3D TRANSLATOR")
+    typer.echo(f"Unity Version:\t {UNITY_VERSION}")
+    typer.echo(f"Please ensure Unity is installed at {UNITY_PATH}")
+    typer.echo(f"IN_DIR:\t\t {in_dir}")
+    typer.echo(f"OUT_DIR:\t {out_dir}")
+    typer.echo()
+
+
+# Ending message
+def farewell():
+    typer.echo()
+    typer.echo("Translation Complete!")
+    exit(0)
+
+
+# Farewell message on error
+def farewell_error():
+    typer.echo()
+    typer.echo("Tranlsation Completed with errors")
+    typer.echo("Please see the command line for a list of issues")
+    exit(1)
+
+
+# Translate a single project
+def translate_project(name: str, project_dir: str, out_dir: str):
+    # Create Unity project and copy original files
+    # unity_dir = os.path.join(out_dir, name)
+    # try:
+    # create_project(unity_dir)
+    # copy_files(project_dir, unity_dir)
+    # add_empty_scene(unity_dir)
+    # except UnityError as e:
+    # handleError(e)
+
+    # Translate xml files in individual threads
+    xml_files = [
+        os.path.join(project_dir, file)
+        for file in os.listdir(project_dir)
+        if file.endswith(".xml")
+    ]
+    for file in xml_files:
+        typer.echo(f"Translating file: {os.path.basename(file)}")
+
+        # Will return a yaml file
+        xml_to_unity(file)
```

Comment:
should this returned output be saved to a list? or does the file get written and we're in this for the side effects?

Reply:
Haven't really gotten that far yet. Which do you think is better?

Reply:
for the cli, you definitely need it to write eventually, it's just a question of how you break down the task into functions.  if it's saving to file, I'd maybe add a `write` to the function name.  If not, I'd have it return some sort of structure that you then have another function actually write.  Both work.

Reply:
I refactored things a little bit. I think I'll build the `Story` dataclass first (renamed the `xml_to_unity` as `read_xml`) and then build the project.  Added issue #13 for now, will probably break into smaller issues once I get to that step.